CMakeLists.txt

```diff
@@ -440,9 +440,12 @@ option(WITH_CEPHFS "CephFS is enabled" ON)
 #option for Mgr
 option(WITH_MGR "ceph-mgr is enabled" ON)
 if(WITH_MGR)
-  set(Python_ADDITIONAL_VERSIONS 2.7)
-  find_package(PythonInterp 2.7 REQUIRED)
-  find_package(PythonLibs 2.7 REQUIRED)
+  set(MGR_PYTHON_VERSION "2.7" CACHE
+    STRING "minimal required version of python runtime for running mgr plugins")
```

Comment:
@tchaikov Cmake is a build system, so if a particular setting is necessary in a py3-only environment, the comment or definition should note that.

I will try the py3-only build again without overriding this value, and if it fails (as I remember it doing), will post the exact error message.

Thanks for suggesting `-DMGR_PYTHON_VERSION=3` - I'll try that as well.

In short, I find this help text "minimal required version of python runtime for running mgr plugins" to be incomplete in the sense that it omits a fact that is important for someone trying to build (not just run) MGR in a py3-only environment.

Reply:
@tchaikov So, the point I'm trying to make is that the current help string says (or implies) that 2.7 is the minimal required version to build. But this only hold true on a system with Python 2 installed. In a Python 3 (only) system, the minimum required version is 3.

That's why I suggested a comment to help folks who might be affected by the build failure because they neglect to override this default (cached) value.

Reply:
please note a a cached variable is a user-facing setting. see https://cmake.org/cmake/help/v3.6/command/set.html#set-cache-entry .

Reply:
OK, so you're saying it should be obvious to the user.

Reply:
When the parameter definition says this specifies the minimum required version, the default is 2.7, and I have 3, then logic dictates that I don't need to change the setting. But in reality I do need to change it. This is not obvious.

Reply:
@smithfarm yes, it should be obvious to a maintainer who's working on cmake projects.

Reply:
Anyway, it's not important - I can add the comment in a separate PR.